android/src/main/java/com/reactcommunity/rndatetimepicker/RNDismissableDatePickerDialog.java

```diff
@@ -62,24 +64,20 @@ protected void onStop() {
     }
   }
 
-  private void fixSpinner(Context context, int year, int month, int dayOfMonth) {
+  private void fixSpinner(Context context, int year, int month, int dayOfMonth, RNDatePickerDisplay display) {
     if (Build.VERSION.SDK_INT == Build.VERSION_CODES.N) {
       try {
         // Get the theme's android:datePickerMode
-        final int MODE_SPINNER = 2;
         Class<?> styleableClass = Class.forName("com.android.internal.R$styleable");
         Field datePickerStyleableField = styleableClass.getField("DatePicker");
         int[] datePickerStyleable = (int[]) datePickerStyleableField.get(null);
 
         final TypedArray a =
             context.obtainStyledAttributes(
                 null, datePickerStyleable, android.R.attr.datePickerStyle, 0);
-        Field datePickerModeStyleableField = styleableClass.getField("DatePicker_datePickerMode");
-        int datePickerModeStyleable = datePickerModeStyleableField.getInt(null);
-        final int mode = a.getInt(datePickerModeStyleable, MODE_SPINNER);
         a.recycle();
 
-        if (mode == MODE_SPINNER) {
+        if (display == RNDatePickerDisplay.SPINNER) {
```

Comment:
@luancurti why does the previous solution not work here? Thanks.

Reply:
That's good question

Reply:
In some cases according the theme o datetime picker the mode is `1` or `2` see this gist https://gist.github.com/jeffdgr8/6bc5f990bf0c13a7334ce385d482af9f#file-timepickerdialogfixednougatspinner-java-L68 and this comments https://gist.github.com/jeffdgr8/6bc5f990bf0c13a7334ce385d482af9f#gistcomment-1888988
https://gist.github.com/jeffdgr8/6bc5f990bf0c13a7334ce385d482af9f#gistcomment-1911283
because this I changed this logic and after some tests on android `7.0` when passing `display = calendar` the picker `always` show `spinner` to `respect the display` I change this comparison to only show this property. What do you think?

Reply:
Alright that makes sense to me

Reply:
@luancurti previously, `fixSpinner` was called from 

**TimePicker**

https://github.com/react-native-community/react-native-datetimepicker/blob/e6d2980230b1e27fa8aed989ccbb79897e6ef5eb/android/src/main/java/com/reactcommunity/rndatetimepicker/RNDismissableTimePickerDialog.java#L43

and **DatePicker**

https://github.com/react-native-community/react-native-datetimepicker/blob/e6d2980230b1e27fa8aed989ccbb79897e6ef5eb/android/src/main/java/com/reactcommunity/rndatetimepicker/RNDismissableDatePickerDialog.java#L42

with your change, because of `if (display == RNDatePickerDisplay.SPINNER)`, it will only be executed for `DatePicker` and not for `TimePicker` because for `TimePicker` there is `RNTimePickerDisplay.SPINNER`, and  `RNTimePickerDisplay.SPINNER != RNDatePickerDisplay.SPINNER`. Is this change intended? 

Is the bug only present on `DatePicker` or is it present on both `DatePicker` and `TimePicker`? Thanks!

Reply:
@vonovak I'll investigate on `TimePicker`, now I don't know 😕

Reply:
@vonovak apparently `TimePicker` has the same `issue` https://github.com/react-native-community/react-native-datetimepicker/blob/d60bff69c54f0180d3373367d5efbd8c2f455744/android/src/main/java/com/reactcommunity/rndatetimepicker/RNDismissableTimePickerDialog.java#L75 but I have an idea to control when show `TimePicker`  we can add a `isVisible` (in `DatePicker` case we should check `display` property is equal `RNDatePickerDisplay.SPINNER`) property to control when the `DatePicker` or `TimePicker` should show and remove this `if` checking `mode === MODE_SPINNER`. What do you think?

Reply:
firstly, in my previous comment I commented on `fixSpinner` being called from `TimePicker` and `DatePicker`, and I thought they were the same functions, but they are not.

You already fixed the `fixSpinner` method in `RNDismissableDatePickerDialog.java`, which means we can add the same logic also to `RNDismissableTimePickerDialog.java` - correct?

> we can add a `isVisible`

sorry, I don't understand what you mean by that / why that is needed.

> remove this if checking mode === MODE_SPINNER 

yes, that check is not needed and this check https://github.com/react-native-community/react-native-datetimepicker/blob/385c9ebf7bd948538e14a92831ec352c0fd87cff/android/src/main/java/com/reactcommunity/rndatetimepicker/RNDismissableDatePickerDialog.java#L80

can be moved up into `if (Build.VERSION.SDK_INT == Build.VERSION_CODES.N)...`

lastly, we should check that switching display from `SPINNER` to some to some other type is correctly reflected by the module, because some of it's state appears to be reused (see https://github.com/react-native-community/react-native-datetimepicker/blob/385c9ebf7bd948538e14a92831ec352c0fd87cff/android/src/main/java/com/reactcommunity/rndatetimepicker/RNTimePickerDialogModule.java#L103). Hope you understand what I mean :)

Reply:
Sorry for the delay

> firstly, in my previous comment I commented on `fixSpinner` being called from `TimePicker` and `DatePicker`, and I thought they were the same functions, but they are not.
> 
> You already fixed the `fixSpinner` method in `RNDismissableDatePickerDialog.java`, which means we can add the same logic also to `RNDismissableTimePickerDialog.java` - correct?
> 

I believe so

> 
> sorry, I don't understand what you mean by that / why that is needed.
> 
> > remove this if checking mode === MODE_SPINNER
> 
> yes, that check is not needed and this check
> 
> https://github.com/react-native-community/react-native-datetimepicker/blob/385c9ebf7bd948538e14a92831ec352c0fd87cff/android/src/main/java/com/reactcommunity/rndatetimepicker/RNDismissableDatePickerDialog.java#L80
> 
> can be moved up into `if (Build.VERSION.SDK_INT == Build.VERSION_CODES.N)...`
> 

Sure!

> lastly, we should check that switching display from `SPINNER` to some to some other type is correctly reflected by the module, because some of its state appears to be reused (see
> 
> https://github.com/react-native-community/react-native-datetimepicker/blob/385c9ebf7bd948538e14a92831ec352c0fd87cff/android/src/main/java/com/reactcommunity/rndatetimepicker/RNTimePickerDialogModule.java#L103
> 
I'll check this. Thanks for detailed explanation!